src/media/js/views/search.js

```diff
@@ -117,6 +117,7 @@ define('views/search',
         };
 
         var query = $('#search-q').val() || $('#search-q-desktop').val();
+        var query = query.replace( /  +/g, ' ' );
```

Comment:
@ddurst I guess this regex will work out for spaces, tabs etc. `str.replace( /\s\s+/g, ' ' )`.